File: 1217.minimum-cost-to-move-chips-to-the-same-position.js
/*
 * @lc app=leetcode id=1217 lang=javascript
 *
 * [1217] Minimum Cost to Move Chips to The Same Position
 */

// @lc code=start
/**
 * @param {number[]} position
 * @return {number}
 */
var minCostToMoveChips = function(position) {
  return Math.min(...position.reduce((a,v)=>{a[v%2]++; return a},[0,0]))
};
// @lc code=end

File: 1217.minimum-cost-to-move-chips-to-the-same-position.py
#
# @lc app=leetcode id=1217 lang=python3
#
# [1217] Minimum Cost to Move Chips to The Same Position
#

# @lc code=start
class Solution:
    def minCostToMoveChips(self, position: List[int]) -> int:
        # min of sum odds / sum events
        odds = len([x for x in position if x%2])
        evens = len([x for x in position if not x%2])
        return min(odds, evens)
# @lc code=end

